### PR TITLE
[WIP] Update sample_calendar.json and database.rules.json

### PR DESCRIPTION
**Pull request by andrayantelo**
Is this a good way to structure it to manage authorization for reading and writing to different calendars?
Addresses issue #107

### CODE REVIEW

File: sample_calendar.json [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
+        "checkit_148460826559529600" { :
+            {
             "startDateString":"20170101",
             "endDateString":"20170202",
             "title":"hi",
             "uniqueId":148460826559529600,
             "checkedDays":{}
             },
+            "permissionRead": "JULxYGaHuFYlSkXqTC2Fx9KIofp2",
+            "permissionWrite": "JULxYGaHuFYlSkXqTC2Fx9KIofp2"
```
Comment (andrayantelo) [inline sample_calendar.json]: I could also do:
"permissionRead" : {
    "JULxYGaHuFYlSkXqTC2Fx9KIofp2" : "true"
}

Although, I'm not sure if that would make sense? I mean, will it ever be false?

File: database.rules.json [hunk 2]
```diff
@@ -2,14 +2,16 @@
...
       }
     },
-    "calendars": {
-      "$calendar_id": {
-      ".read": "auth !=null && data.child('permissionRead').hasChild(root.child('users/'+auth.uid).val())",
-      ".write": "auth !=null && data.child('permissionWrite/' + auth.uid.val()).exists()"
+    "calendars":{
+      "$calendarId": {
+        "calendarState": {
+          ".read" : "auth!== null && data.parent().child('read').hasChild(auth.uid) && data.parent().child('read').child(auth.uid).val() === true",
```
Comment (johntyree) [inline database.rules.json]: Two things here.

First, you shouldn't ever need to do `=== true`. Think about why.

Second, I think the value of the key doesn't really matter anyway. You're not going to put in uids with values that are anything other than `true` right? What about `".read": "auth !== null && data.parent().child('read/' + auth.uid).exists()"` ?

**Review by johntyree — CHANGES_REQUESTED** [review]

File: database.rules.json [hunk 3]
```diff
@@ -9,9 +9,17 @@
...
-          ".read" : "auth!== null && data.parent().child('read').hasChild(auth.uid) && data.parent().child('read').child(auth.uid).val() === true",
-          ".write" : "auth !== null && data.parent().child('write').hasChild(auth.uid) && data.parent().child('write').child(auth.uid).val() === true"
-        } 
+          ".read" : "auth!== null && data.parent().child('read/' + auth.uid).exists()",
+          ".write" : "auth !== null && data.parent().child('write/' + auth.uid).exists()"
+        },
+        "write" : {
+          ".read" :,
+          ".write" :
+        }
```
Comment (johntyree) [inline database.rules.json]: Missing a comma

File: database.rules.json [hunk 4]
```diff
@@ -9,9 +9,17 @@
     "calendars":{
       "$calendarId": {
         "calendarState": {
-          ".read" : "auth!== null && data.parent().child('read').hasChild(auth.uid) && data.parent().child('read').child(auth.uid).val() === true",
-          ".write" : "auth !== null && data.parent().child('write').hasChild(auth.uid) && data.parent().child('write').child(auth.uid).val() === true"
-        } 
+          ".read" : "auth!== null && data.parent().child('read/' + auth.uid).exists()",
```
Comment (johntyree) [inline database.rules.json]: Missing a space after first `auth`

**Review by johntyree — CHANGES_REQUESTED** [review]
https://jsonformatter.curiousconcept.com/

**Review by johntyree — APPROVED** [review]

**Pull request merged**